notifications-kibana-plugin/public/pages/Emails/components/tables/RecipientGroupsTable.tsx

```diff
@@ -0,0 +1,256 @@
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import {
+  RecipientGroupItemType,
+  TableState,
+} from '../../../../../models/interfaces';
+import { Component } from 'react';
+import { CoreServicesContext } from '../../../../components/coreServices';
+import {
+  EuiBasicTable,
+  EuiButton,
+  EuiEmptyPrompt,
+  EuiFieldSearch,
+  EuiHorizontalRule,
+  EuiTableFieldDataColumnType,
+  EuiTableSortingType,
+} from '@elastic/eui';
+import { SORT_DIRECTION } from '../../../../../common';
+import { ROUTES } from '../../../../utils/constants';
+import React from 'react';
+import { Criteria } from '@elastic/eui/src/components/basic_table/basic_table';
+import { Pagination } from '@elastic/eui/src/components/basic_table/pagination_bar';
+import { DEFAULT_PAGE_SIZE_OPTIONS } from '../../../Notifications/utils/constants';
+import {
+  ContentPanel,
+  ContentPanelActions,
+} from '../../../../components/ContentPanel';
+import { ModalConsumer } from '../../../../components/Modal';
+import { DeleteSenderModal } from '../modals/DeleteSenderModal';
+
+interface RecipientGroupsTableProps {}
+
+interface RecipientGroupsTableState
+  extends TableState<RecipientGroupItemType> {}
+
+export class RecipientGroupsTable extends Component<
+  RecipientGroupsTableProps,
+  RecipientGroupsTableState
+> {
+  static contextType = CoreServicesContext;
+  columns: EuiTableFieldDataColumnType<RecipientGroupItemType>[];
+
+  constructor(props: RecipientGroupsTableProps) {
+    super(props);
+
+    this.state = {
+      total: 1,
+      from: 0,
+      size: 5,
+      search: '',
+      sortField: 'name',
+      sortDirection: SORT_DIRECTION.ASC,
+      items: Array.from({ length: 5 }, (v, i) => ({
+        id: i.toString(),
+        name: 'Group ' + (i + 1),
+        email: new Array(Math.round(Math.random() * 5) + 1).fill({
+          email: 'no-reply@company.com',
+        }),
+        description: 'Description ' + i,
+      })),
+      selectedItems: [],
+      loading: true,
+    };
+
+    this.columns = [
+      {
+        field: 'name',
+        name: 'Name',
+        sortable: true,
+        truncateText: true,
+        width: '150px',
+      },
+      {
+        field: 'email',
+        name: 'Email addresses',
+        sortable: true,
+        truncateText: true,
+        width: '150px',
+        render: (emails: string[]) => emails.length,
+      },
+      {
+        field: 'description',
+        name: 'Description',
+        sortable: true,
+        truncateText: true,
+        width: '150px',
+      },
+    ];
+  }
+
+  async componentDidMount() {
+    // await this.getNotifications();
```

Comment:
Can we remove `componentDidMount` if we are not calling this statement?

Reply:
changed to TODO comment